docs/tendermint-core/state-id.md

```diff
@@ -39,5 +39,18 @@ graph TD
 
 As a result:
 
-* verification of a vote for the block at height `N` requires information about the state at height `N-1`
-* block at height `N` contains a commit referring to state ID at height `N-2`
+* verification of a vote for the block at height `N` requires information about the AppHash returned from ABCI App in response to a commit for the height `N-1`,
+* block at height `N` contains the commit referring to state ID at height `N-2`.
```

Comment:
I guess this is exactly because of it doesn’t work in Drive.

Reply:
```
StateID(N-2) ={ LastAppHash: Block(N-1).AppHash , Height: N-2 }
```
And it's part of Commit(Height=N-1), and signed with the same keys as this commit

Reply:
But we don't have this commit on N, right. Only on N+1.